src/Clients/BaseClient.php

```diff
@@ -5,11 +5,33 @@
 namespace Creso\LaravelEasyofficeApi\Clients;
 
 use Illuminate\Http\Client\PendingRequest;
+use Illuminate\Http\Client\Response;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Validation\ValidationException;
 
 class BaseClient
 {
     public function __construct(protected PendingRequest $httpClient)
     {
+        $this->httpClient->throw([$this, 'handleErrors']);
+    }
+
+    public function getRequest(string $endpoint, array $options = [], bool $skipCache = false): array
+    {
+        $cacheKey = $this->generateCacheKey($endpoint);
+
+        if ($this->getCacheEnabled() && Cache::has($cacheKey) && ! $skipCache) {
```

Comment:
I would probably extract this to a `shouldCache` method to make it a bit more readable